--- a	2023-09-28 18:47:15.484014461 +0100
+++ b	2023-09-28 18:47:15.504015194 +0100
@@ -87,27 +87,6 @@
    and with_scope ~scope lam =
      let old_scope = !current_scope in
      current_scope := scope;
-diff --git a/testsuite/tests/backtrace/backtrace_dynlink.flambda.reference b/testsuite/tests/backtrace/backtrace_dynlink.flambda.reference
---- a/testsuite/tests/backtrace/backtrace_dynlink.flambda.reference
-+++ b/testsuite/tests/backtrace/backtrace_dynlink.flambda.reference
-@@ -2,7 +2,7 @@ Raised by primitive operation at Backtrace_dynlink_plugin in file "backtrace_dyn
- Called from Dynlink.Native.run.(fun) in file "otherlibs/dynlink/native/dynlink.ml", line 87, characters 12-29
- Called from Stdlib__List.iter in file "list.ml" (inlined), line 112, characters 12-15
- Called from Dynlink.Native.run in file "otherlibs/dynlink/native/dynlink.ml", line 86, characters 4-273
--Called from Dynlink_common.Make.load.(fun) in file "otherlibs/dynlink/dynlink_common.ml" (inlined), line 363, characters 13-56
-+Called from Dynlink_common.Make.load.(fun) in file "otherlibs/dynlink/dynlink_common.ml", line 363, characters 13-56
- Called from Stdlib__List.iter in file "list.ml" (inlined), line 112, characters 12-15
- Called from Dynlink_common.Make.load in file "otherlibs/dynlink/dynlink_common.ml", line 359, characters 8-392
- Called from Dynlink_common.Make.loadfile in file "otherlibs/dynlink/dynlink_common.ml" (inlined), line 374, characters 26-45
-@@ -12,7 +12,7 @@ Raised by primitive operation at Dynlink.Native.run.(fun) in file "otherlibs/dyn
- Re-raised at Dynlink.Native.run.(fun) in file "otherlibs/dynlink/native/dynlink.ml", line 89, characters 10-149
- Called from Stdlib__List.iter in file "list.ml" (inlined), line 112, characters 12-15
- Called from Dynlink.Native.run in file "otherlibs/dynlink/native/dynlink.ml", line 86, characters 4-273
--Called from Dynlink_common.Make.load.(fun) in file "otherlibs/dynlink/dynlink_common.ml" (inlined), line 363, characters 13-56
-+Called from Dynlink_common.Make.load.(fun) in file "otherlibs/dynlink/dynlink_common.ml", line 363, characters 13-56
- Called from Stdlib__List.iter in file "list.ml" (inlined), line 112, characters 12-15
- Called from Dynlink_common.Make.load in file "otherlibs/dynlink/dynlink_common.ml", line 359, characters 8-392
- Re-raised at Dynlink_common.Make.load in file "otherlibs/dynlink/dynlink_common.ml", line 372, characters 8-17
 diff --git a/testsuite/tests/backtrace/lazy.flambda.reference b/testsuite/tests/backtrace/lazy.flambda.reference
 new file mode 100644
 --- /dev/null
@@ -115,14 +94,14 @@
 @@ -0,0 +1,12 @@
 +Uncaught exception Not_found
 +Raised at Lazy.l1 in file "lazy.ml", line 3, characters 28-45
-+Called from CamlinternalLazy.do_force_block in file "camlinternalLazy.ml", line 49, characters 17-27
-+Re-raised at CamlinternalLazy.do_force_block in file "camlinternalLazy.ml", line 56, characters 4-11
++Called from CamlinternalLazy.force_lazy_block in file "camlinternalLazy.ml", line 119, characters 17-27
++Re-raised at CamlinternalLazy.force_lazy_block in file "camlinternalLazy.ml", line 124, characters 4-11
 +Called from Lazy.test1 in file "lazy.ml", line 6, characters 11-24
 +Called from Lazy.run in file "lazy.ml", line 15, characters 4-11
 +Uncaught exception Not_found
 +Raised at Lazy.l2 in file "lazy.ml", line 8, characters 28-45
-+Called from CamlinternalLazy.do_force_block in file "camlinternalLazy.ml", line 49, characters 17-27
-+Re-raised at CamlinternalLazy.do_force_block in file "camlinternalLazy.ml", line 56, characters 4-11
++Called from CamlinternalLazy.force_lazy_block in file "camlinternalLazy.ml", line 119, characters 17-27
++Re-raised at CamlinternalLazy.force_lazy_block in file "camlinternalLazy.ml", line 124, characters 4-11
 +Called from Lazy.test2 in file "lazy.ml", line 11, characters 6-15
 +Called from Lazy.run in file "lazy.ml", line 15, characters 4-11
 diff --git a/testsuite/tests/backtrace/lazy.ml b/testsuite/tests/backtrace/lazy.ml
@@ -157,13 +136,12 @@
 diff --git a/testsuite/tests/backtrace/lazy.reference b/testsuite/tests/backtrace/lazy.reference
 --- a/testsuite/tests/backtrace/lazy.reference
 +++ b/testsuite/tests/backtrace/lazy.reference
-@@ -1,14 +1,14 @@
+@@ -1,12 +1,12 @@
  Uncaught exception Not_found
 -Raised at Lazy.l1 in file "lazy.ml", line 7, characters 28-45
 +Raised at Lazy.l1 in file "lazy.ml", line 3, characters 28-45
- Called from CamlinternalLazy.do_force_block in file "camlinternalLazy.ml", line 49, characters 17-27
- Re-raised at CamlinternalLazy.do_force_block in file "camlinternalLazy.ml", line 56, characters 4-11
- Called from CamlinternalLazy.force_lazy_block in file "camlinternalLazy.ml" (inlined), line 78, characters 27-67
+ Called from CamlinternalLazy.force_lazy_block in file "camlinternalLazy.ml", line 119, characters 17-27
+ Re-raised at CamlinternalLazy.force_lazy_block in file "camlinternalLazy.ml", line 124, characters 4-11
 -Called from Lazy.test1 in file "lazy.ml", line 10, characters 11-24
 -Called from Lazy.run in file "lazy.ml", line 19, characters 4-11
 +Called from Lazy.test1 in file "lazy.ml", line 6, characters 11-24
@@ -171,9 +149,8 @@
  Uncaught exception Not_found
 -Raised at Lazy.l2 in file "lazy.ml", line 12, characters 28-45
 +Raised at Lazy.l2 in file "lazy.ml", line 8, characters 28-45
- Called from CamlinternalLazy.do_force_block in file "camlinternalLazy.ml", line 49, characters 17-27
- Re-raised at CamlinternalLazy.do_force_block in file "camlinternalLazy.ml", line 56, characters 4-11
- Called from CamlinternalLazy.force_lazy_block in file "camlinternalLazy.ml" (inlined), line 78, characters 27-67
+ Called from CamlinternalLazy.force_lazy_block in file "camlinternalLazy.ml", line 119, characters 17-27
+ Re-raised at CamlinternalLazy.force_lazy_block in file "camlinternalLazy.ml", line 124, characters 4-11
 -Called from Lazy.test2 in file "lazy.ml", line 15, characters 6-15
 -Called from Lazy.run in file "lazy.ml", line 19, characters 4-11
 +Called from Lazy.test2 in file "lazy.ml", line 11, characters 6-15
